tests/detached_editing/test_serialization.py

```diff
@@ -0,0 +1,139 @@
+import json
+import unittest
+from base64 import b64decode, b64encode
+
+from qgis.core import QgsGeometry
+from qgis.PyQt.QtCore import QDate, QDateTime, QTime
+
+from nextgis_connect.detached_editing.serialization import (
+    deserialize_geometry,
+    deserialize_value,
+    serialize_geometry,
+    serialize_value,
+)
+from tests.ng_connect_testcase import (
+    NgConnectTestCase,
+)
+
+
+class TestSerialization(NgConnectTestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.types = [
+            (12345, "12345", 12345),  # integer
+            (8388607, "8388607", 8388607),  # mediumint
+            (3.14159, "3.14159", 3.14159),  # real
+            ("Hello, World!", '"Hello, World!"', "Hello, World!"),  # text
+            (
+                QDate(2025, 3, 4),
+                json.dumps("2025-03-04"),
+                "2025-03-04",
+            ),  # date
+            (QTime(14, 30, 15), json.dumps("14:30:15"), "14:30:15"),  # time
```

Comment:
Please also add tests for types from the datetime library. It will require some code modifications.